### PR TITLE
Reformat SASS files with prettier

### PR DESCRIPTION
**Pull request by blowery**
The results of running prettier over all of the scss files in the repo.

#### Changes proposed in this Pull Request

* Reformat scss files with prettier

#### Testing instructions

* Everything should look the way it did before
* CSS output shouldn't really change

### CODE REVIEW

File: client/components/popover/style.scss [hunk 1]
```diff
@@ -7,15 +7,14 @@
 	z-index: z-index( 'root', '.popover' );
 	position: absolute;
 	top: 0;
-	left: 0 #{"/*rtl:ignore*/"};
-	right: auto #{"/*rtl:ignore*/"};
+	left: 0 #{'/*rtl:ignore*/'};
```
Comment (blowery) [inline client/components/popover/style.scss]: I'm pretty sure this is using interpolation to insert an actual comment for later processing by rtlcss. I don't think moving from " to ' has any effect.

**Comment by blowery** [comment]
Added #29604 as an optional add-on to this PR, which updates our stylelint rules a touch and runs `--fix` on everything.

**Comment by blowery** [comment]
Hrm, turns out prettier's SASS support has a few nasty bugs:
https://github.com/prettier/prettier/issues?utf8=%E2%9C%93&q=is%3Aissue+is%3Aopen+scss

Maybe we don't want to use prettier for SASS just yet...?

**Comment by blowery** [comment]
To repro the bugs in our codebase, run prettier against `assets/stylesheets/shared/functions/_z-index.scss`

**Comment by blowery** [comment]
Closing in favor of #29697

**Comment by belcherj** [comment]
Prettier tanks on:  assets/stylesheets/shared/functions/_z-index.scss

**Comment by jsnajdr** [comment]
I spent some time hacking Prettier over the holidays, and came up with two fixes for SCSS formatting:
- shellscape/postcss-values-parser#60
- prettier/prettier#5710

Some other bugs still remain, including some that break code rather than just formatting it incorrectly. The most severe is prettier/prettier#5603. There I posted an analysis of the causes, but a fix won't be easy.

**Pull request closed without merging**